recipes-core/tdx-enc-handler/tdx-enc-handler/tdx-enc-handler.service

```diff
@@ -1,6 +1,8 @@
 [Unit]
 Description=Encryption handler for Toradex modules
+DefaultDependencies=no
```

Comment:
For the default BSP image, it works without this line, so it is just an optimization. But since the final user might have early-boot services that might depend on the encrypted data partition, the soon we run this, the better.

Looking at the default early-boot systemd services, most of them (more than 60) set this option, so I thought we should do the same:

```
# grep -R DefaultDependencies=no /lib/systemd/system/*.service | wc -l
66

# grep -R DefaultDependencies=no /lib/systemd/system/*.service
/lib/systemd/system/debug-shell.service:DefaultDependencies=no  
/lib/systemd/system/emergency.service:DefaultDependencies=no          
/lib/systemd/system/initrd-cleanup.service:DefaultDependencies=no              
/lib/systemd/system/initrd-parse-etc.service:DefaultDependencies=no   
/lib/systemd/system/initrd-switch-root.service:DefaultDependencies=no
/lib/systemd/system/initrd-udevadm-cleanup-db.service:DefaultDependencies=no
/lib/systemd/system/kmod-static-nodes.service:DefaultDependencies=no    
/lib/systemd/system/ldconfig.service:DefaultDependencies=no       
/lib/systemd/system/modprobe@.service:DefaultDependencies=no         
/lib/systemd/system/pam_namespace.service:DefaultDependencies=no   
/lib/systemd/system/quotaon.service:DefaultDependencies=no           
/lib/systemd/system/rescue.service:DefaultDependencies=no          
/lib/systemd/system/rpcbind.service:DefaultDependencies=no
/lib/systemd/system/system-update-cleanup.service:DefaultDependencies=no                                                                                                                                           
/lib/systemd/system/systemd-ask-password-console.service:DefaultDependencies=no
/lib/systemd/system/systemd-backlight@.service:DefaultDependencies=no
/lib/systemd/system/systemd-exit.service:DefaultDependencies=no
/lib/systemd/system/systemd-fsck-root.service:DefaultDependencies=no
/lib/systemd/system/systemd-fsck@.service:DefaultDependencies=no
/lib/systemd/system/systemd-halt.service:DefaultDependencies=no
/lib/systemd/system/systemd-hibernate-resume@.service:DefaultDependencies=no
/lib/systemd/system/systemd-hibernate.service:DefaultDependencies=no
/lib/systemd/system/systemd-hwdb-update.service:DefaultDependencies=no
/lib/systemd/system/systemd-hybrid-sleep.service:DefaultDependencies=no
/lib/systemd/system/systemd-initctl.service:DefaultDependencies=no
/lib/systemd/system/systemd-journal-catalog-update.service:DefaultDependencies=no
/lib/systemd/system/systemd-journal-flush.service:DefaultDependencies=no
/lib/systemd/system/systemd-journald.service:DefaultDependencies=no
/lib/systemd/system/systemd-kexec.service:DefaultDependencies=no
/lib/systemd/system/systemd-machine-id-commit.service:DefaultDependencies=no
/lib/systemd/system/systemd-modules-load.service:DefaultDependencies=no
/lib/systemd/system/systemd-network-generator.service:DefaultDependencies=no
/lib/systemd/system/systemd-networkd-wait-online.service:DefaultDependencies=no
/lib/systemd/system/systemd-networkd.service:DefaultDependencies=no
/lib/systemd/system/systemd-poweroff.service:DefaultDependencies=no
/lib/systemd/system/systemd-pstore.service:DefaultDependencies=no
[...]
```

Anyway, I don't have any benchmark on if setting this line makes the boot faster or not.

Reply:
To be honest, I'm not sure why a dependency loop error/warning is not triggered without `DefaultDependencies=no`, as I mentioned in the previous PR (https://github.com/toradex/meta-toradex-security/pull/18#discussion_r1569731410).

Anyways, I think disabling default dependencies is the proper setting for us.